packages/cardpay-sdk/sdk/scheduled-payment-module.ts

```diff
@@ -0,0 +1,147 @@
+import Web3 from 'web3';
+import GnosisSafeABI from '../contracts/abi/gnosis-safe';
+import MetaGuardABI from '../contracts/abi/modules/meta-guard';
+import ScheduledPaymentABI from '../contracts/abi/modules/scheduled-payment-module';
+import { getAddress } from '../contracts/addresses';
+import { AbiItem, fromWei } from 'web3-utils';
+import { deployAndSetUpModule, encodeMultiSend } from './utils/module-utils';
+import { TransactionOptions, waitForTransactionConsistency } from './utils/general-utils';
+import { ContractOptions } from 'web3-eth-contract';
+import { executeTransaction, gasEstimate, getNextNonceFromEstimate, Operation } from './utils/safe-utils';
+import { Signer } from 'ethers';
+import { signSafeTxAsRSV } from './utils/signing-utils';
+import { BN } from 'bn.js';
+import { ERC20ABI } from '..';
+import { ZERO_ADDRESS } from './constants';
+
+export default class ScheduledPaymentModule {
+  constructor(private layer2Web3: Web3, private layer2Signer?: Signer) {}
+
+  async enableModule(
+    safeAddress: string,
+    gasTokenAddress: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
+  ) {
+    if (!safeAddress) {
+      throw new Error('safeAddress must be specified');
+    }
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    let from = contractOptions?.from ?? (await this.layer2Web3.eth.getAccounts())[0];
+
+    let enableModuleTxs = await this.generateEnableModuleTxs(safeAddress);
+    let setGuardTxs = await this.generateSetGuardTxs(safeAddress);
+
+    let multiSendTransaction = await encodeMultiSend(this.layer2Web3, [...enableModuleTxs.txs, ...setGuardTxs.txs]);
+
+    let estimate = await gasEstimate(
+      this.layer2Web3,
+      safeAddress,
+      multiSendTransaction.to,
+      multiSendTransaction.value,
+      multiSendTransaction.data,
+      multiSendTransaction.operation,
+      gasTokenAddress
+    );
+    let gasCost = new BN(estimate.safeTxGas).add(new BN(estimate.baseGas)).mul(new BN(estimate.gasPrice));
+
+    let token = new this.layer2Web3.eth.Contract(ERC20ABI as AbiItem[], gasTokenAddress);
+    let symbol = await token.methods.symbol().call();
+    let balance = new BN(await token.methods.balanceOf(safeAddress).call());
+    if (balance.lt(gasCost)) {
+      throw new Error(
+        `Safe does not have enough balance to enable scheduled payment module. The gas token ${gasTokenAddress} balance of the safe ${safeAddress} is ${fromWei(
+          balance
+        )}, the the gas cost is ${fromWei(gasCost)} ${symbol}`
+      );
+    }
+    if (nonce == null) {
+      nonce = getNextNonceFromEstimate(estimate);
+      if (typeof onNonce === 'function') {
+        onNonce(nonce);
+      }
+    }
+    let signatures = await signSafeTxAsRSV(
```

Comment:
Let's use `signSafeTX` from the cardpay-sdk/sdk/utils/signing-utils.ts module--it's more concise

Reply:
Updated